code/src/components/app-range-slider/AppRangeSlider.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { screen } from "@testing-library/react";
+
+import AppRangeSlider from "@/components/app-range-slider/AppRangeSlider";
+
+import renderWithProviders from "@/utils/render-with-providers/renderWithProviders";
+import typeIntoInput from "@/utils/type-into-input/typeIntoInput";
+
+const mockHandleChange = jest.fn();
+
+describe("AppRangeSlider", () => {
+  describe("with default props", () => {
+    test("uses default props correctly", () => {
+      renderWithProviders(<AppRangeSlider />);
+
+      const rangeStartInput = screen.getByTestId("range-start");
+      expect(rangeStartInput).toHaveAttribute("min", "0");
+      expect(rangeStartInput).toHaveAttribute("max", "20000");
+      expect(rangeStartInput).toHaveAttribute("step", "10");
+    });
+  });
+
+  describe("with ordinary props", () => {
+    beforeEach(() => {
+      renderWithProviders(
+        <AppRangeSlider
+          value={[10, 40]}
+          min={0}
+          max={100}
+          step={10}
+          onChange={mockHandleChange}
+        />
+      );
+    });
```

Comment:
`renderWithProviders` repeated for now. As I told create own render function
You did it before and call inside of tests

<img width="742" alt="image" src="https://github.com/user-attachments/assets/63e44e5e-51cc-448e-927b-51450e21114f">

Reply:
I tried to do it, but it does not make much sense because props differ very much. A intentionally created two `describe` blocks to take advantage of `beforeEach` hook for solving code reusability problem

![image](https://github.com/user-attachments/assets/428b6161-81f0-4714-8c79-38a060970008)

Reply:
> I tried to do it, but it does not make much sense because props differ very much. A intentionally created two `describe` blocks to take advantage of `beforeEach` hook for solving code reusability problem
> 
> ![image](https://private-user-images.githubusercontent.com/109922101/348619151-428b6161-81f0-4714-8c79-38a060970008.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwMjY5NjMsIm5iZiI6MTcyMTAyNjY2MywicGF0aCI6Ii8xMDk5MjIxMDEvMzQ4NjE5MTUxLTQyOGI2MTYxLTgxZjAtNDcxNC04Yzc5LTM4YTA2MDk3MDAwOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNVQwNjU3NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZjJlNmFhMzdlYWM5ZDFiZGMzMDQ2MWViMzU5OWU0MWU1YmUwOGQzNTI5YmE5ZDNiMzA2ZTg5OTEyZDg3MjgwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kOtULyceDfE0zvwYeZerU-xb0g7DqQyQH051rQ5WGbY)

Ok. but try to write with your single approach with `renderComponent` or not. Just differences with your style code logic in separate cases

Reply:
I agree with you, but in different cases we should pick the best approach to solve the problem. You wouldn't use a screwdriver to hammer a nail, the same is true here